README.md
Google sitemaps for TYPO3
======================

This is a "dd_googlesitemap" TYPO3 extension. You will need a TYPO3 CMS to run it.

What does it do?
----------------
This extension adds sitemaps for pages and tt_news items. Typically you add the result to Google like http://example.com/index.php?eID=dd_googlesitemap.

Make sure to read the manual in the doc/ directory before asking questions or writing bug reports. Anything, which is explained in the manual, will be silently closed without answering.

What it does not do?
--------------------
Many things. It is deliberately not a "all-purpose solution". It is created to be fast and efficient. It only contains sitemaps for pages and tt_news and there are no plans to change that or do any sophisticated filtering, etc. You still can ask though :)

Repository
----------
Main repository is at https://github.com/dmitryd/typo3_dd_googlesitemap.

Contact
-------
E-mail: dmitry.dulepov@gmail.com


ChangeLog
2014-03-07  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Released version 1.3.1

2014-03-05  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Fixed #27: version 1.3.0 breaks scheduler in TYPO3 6.1.7
	* Fixed incorrect usage of a flash message queue in TYPO3 6.x

2014-02-14  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Fix scheduler additional field provider for TYPO3 6.x
	* Fixed most of PhpStorm warnings
	* Fixed #26: [FEATURE] Configurable list of non-visible page types
	* Fixed #16: "Backend User Section" is included into sitemap
	* Fixed #10: lastmod is not up-to-date
	* Fixed #4: Make tt_news link object configurable via typoscript
	* Released version 1.3.0

2014-02-11  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Merged various pull requests from github
	* Fixed #14: Priority Number Format

2013-04-23  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Fixed #12: wrong path to the news renderer
	* Fixed wrong 400 header line (thanks to Sascha Hepp)

2013-04-11  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Released version 1.2.0

2013-04-05  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Fixed #3: endless - loop when a extension-specific type ignores offset and limit
	* Updated copyright information

2013-04-01  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Add inspection comments to lower the amount of warnings in PhpStorm

2013-03-26  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Reduce number of update calls to prevent to many updates of the same page in a single request (for example, during copying of pages)

2013-03-07  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Released version 1.0.8

		Happy birthday to me!

2013-03-05  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Snowflake task 2697.65: add an index to sitemaps

2013-02-20  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Fixed #1: Hooking into the list of pages (thanks to Xavier Perseguers)
	* Fixed forge issue #9453: Not all pages get into the sitemap

2013-02-19  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Add limit & offset URL parameters to allow multiple sitemaps
	* Fixed issue #3063: add multilanguage tt_news support
	* Fixed issue #4025: include pages from sysfolders into the sitemap (thanks to Stefan Galinski)
	* Fixed issue #4881: add information about rewrite rules to the manual

2012-12-04  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Ensure compatibility with TYPO3 6.0
	* Released version 1.0.7

2012-02-20  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Fixed bug #34123: Rewrite rule does not work when realurl is enabled
	* Released version 1.0.6

2010-05-19  Dmitry Dulepov  <dmitry.dulepov+typo3svn@gmail.com>

	* Fixed bug #6689: do not limit normal tt_news sitemap to two days
	* Fixed bug #6548: News sitemap lacks content of <news:title> tag
	* Fixed bug #6567: News sitemap: publication name is not configurable
	* Fixed bug #4023: Missing doktype property of the first page in the pageList array
	* Fixed bug #2882: add sitemap autodiscovery information to the manual
	* Released version 1.0.5

2010-01-09  Dmitry Dulepov  <dmitry.dulepov+typo3svn@gmail.com>

	* Fixed: deprecated calls
	* Fixed bug #4157: Annoying logging of changes to tx_ddgooglesitemap_lastmod
	* Fixed bug #3033: News sitemap: respect Google requirements
	* Fixed bug #3716: pages below a shortcut page aren't rendered
	* Released version 1.0.4

2010-01-09  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Fix: wrong XCLASS in TCEmain hook

2009-04-01  Dmitry Dulepov  <dmitry@typo3.org>

	* Prevent duplicate timestamps in the lastmod field

2009-03-18  Dmitry Dulepov  <dmitry@typo3.org>

	* Fixed problem: doktype and no_search were not available
	* Released version 1.0.3

2009-03-17  Dmitry Dulepov  <dmitry@typo3.org>

	* Fixed bug #2862: tt_news sitemap should produce a page sitemap
	* Fixed bug #2704: Shortcut on top level messes up the whole sitemap
	* Fixed bug #2693: Page type Shortcut adds second entry for page
	* Fixed bug #2692: Pages of type "Visual menu separator" are included in sitemap
	* Fixed bug #2867: Exclude 'no search' pages from the sitemap
	* Released version 1.0.1
	* Apply 'forceStartPid' to news sitemaps too
	* Released version 1.0.2

2009-02-25  Dmitry Dulepov  <dmitry@typo3.org>

	* Added an option to skip root line check for news
	* Added documentation
	* Clean up before uploading to TER
	* Released version 1.0.0
	* Fixed bug #2691: GPvar is deprecated in 4.3

2009-01-12  Dmitry Dulepov  <dmitry@typo3.org>

	* Fixed: missing <url> tag in the news sitemap

2008-12-12  Dmitry Dulepov  <dmitry@typo3.org>

	* Initial code with sitemap for pages
	* Added news sitemap
	* Fixed: current page ID was not determined correctly with RealURL
	* Changed news item sorting in news sitemap (newer items first)
	* Added check for 0 when validating storage page id


ext_emconf.php
<?php

/*********************************************************************
* Extension configuration file for ext "dd_googlesitemap".
*
* Generated by ext 14-02-2014 09:05 UTC
*
* https://github.com/t3elmar/Ext
*********************************************************************/

$EM_CONF[$_EXTKEY] = array (
  'title' => 'Google sitemap',
  'description' => 'High performance Google sitemap implementation that avoids typical errors by other similar extensions',
  'category' => 'fe',
  'author' => 'Dmitry Dulepov',
  'author_email' => 'dmitry.dulepov@gmail.com',
  'shy' => '',
  'dependencies' => '',
  'conflicts' => '',
  'priority' => '',
  'module' => '',
  'state' => 'stable',
  'internal' => '',
  'uploadfolder' => 0,
  'createDirs' => 'typo3temp/dd_googlesitemap',
  'modify_tables' => 'pages',
  'clearCacheOnLoad' => 0,
  'lockType' => '',
  'author_company' => 'SIA "ACCIO"',
  'version' => '2.0.0',
  'constraints' =>
  array (
    'depends' =>
    array (
      'typo3' => '6.2.0-6.2.99',
    ),
    'conflicts' =>
    array (
    ),
    'suggests' =>
    array (
    ),
  ),
  '_md5_values_when_last_written' => 'a:19:{s:9:"ChangeLog";s:4:"822f";s:9:"README.md";s:4:"ee4d";s:32:"class.tx_ddgooglesitemap_eid.php";s:4:"d1c3";s:34:"class.tx_ddgooglesitemap_pages.php";s:4:"51cd";s:36:"class.tx_ddgooglesitemap_tcemain.php";s:4:"aedb";s:35:"class.tx_ddgooglesitemap_ttnews.php";s:4:"c00f";s:16:"ext_autoload.php";s:4:"aff4";s:12:"ext_icon.gif";s:4:"0709";s:17:"ext_localconf.php";s:4:"b549";s:14:"ext_tables.php";s:4:"eada";s:14:"ext_tables.sql";s:4:"239c";s:24:"ext_typoscript_setup.txt";s:4:"71ba";s:13:"locallang.xml";s:4:"f8ad";s:14:"doc/manual.sxw";s:4:"1e09";s:56:"renderers/class.tx_ddgooglesitemap_abstract_renderer.php";s:4:"0539";s:52:"renderers/class.tx_ddgooglesitemap_news_renderer.php";s:4:"fcb9";s:54:"renderers/class.tx_ddgooglesitemap_normal_renderer.php";s:4:"ec95";s:63:"scheduler/class.tx_ddgooglesitemap_additionalfieldsprovider.php";s:4:"4267";s:48:"scheduler/class.tx_ddgooglesitemap_indextask.php";s:4:"cbaa";}',
  'suggests' =>
  array (
  ),
  'user' => 'dmitry',
  'comment' => 'Bug fixes + TYPO3 6.2 compatibility + typolink for tt_news',
);

?>